hack/devtools/platform_msi.sh

```diff
@@ -0,0 +1,79 @@
+#!/bin/bash
+
+# This script creates a platform Identities to use for local development
+# The script reads the env var PLATFORM_WORKLOAD_IDENTITY_ROLE_SETS and creates platform identities for each operator
+
+get_platform_workloadIdentity_role_sets() {
+    local platformWorkloadIdentityRole
+   
+    # Parse the JSON data using jq
+    platformWorkloadIdentityRoles=$(echo "${PLATFORM_WORKLOAD_IDENTITY_ROLE_SETS}" | jq -c '.[].platformWorkloadIdentityRoles[]')
+
+    echo "${platformWorkloadIdentityRoles}"
+}
+
+assign_role_to_platform_identity() {
+    local principalId=$1
+    local roleId=$2
+    local scope="/subscriptions/${AZURE_SUBSCRIPTION_ID}/resourceGroups/${RESOURCEGROUP}"
+
+    echo "INFO: Assigning roles to platform identity: ${principalId}"
+    local result=$(az role assignment create --assignee "${principalId}" --role "${roleId}"  --scope "${scope}" --output json)
+
+    echo "Role assignment result: $result"
+    echo ""
+}
+
+create_platform_identity_and_assign_role() {
+    local operatorName="${1}"
+    local roleDefinitionId="${2}"
+
+    echo "INFO: Creating platform identity for operator: ${operatorName}"
+    local result=$(az identity create --name "${operatorName}" --resource-group "${RESOURCEGROUP}" --subscription "${AZURE_SUBSCRIPTION_ID}" --output json)
+
+    # Extract the client ID, principal Id, resource ID and name from the result
+    clientID=$(echo $result | jq -r .clientId)
+    principalId=$(echo $result | jq -r .principalId)
+    resourceId=$(echo $result | jq -r .id)
+    name=$(echo $result | jq -r .name)
+
+    echo "Client ID: $clientID"
+    echo "Principal ID: $principalId"
+    echo "Resource ID: $resourceId"
+    echo "Name: $name"
+    echo ""
+
+    if [[ "${operatorName}" == "MachineApiOperator" || "${operatorName}" == "NetworkOperator" \
+        || "${operatorName}" == "AzureFilesStorageOperator" || "${operatorName}" == "ServiceOperator" ]]; then
```

Comment:
I don't know if this `if` statement should be included:
- The operator names in the platform workload identity role set env var should be allowed to change, this hardcodes them
- There are 8 total roles that need to be assigned to 8 identities, and there are only 4 here - don't we want to assign roles to all identities?

Reply:
>There are 8 total roles that need to be assigned to 8 identities, and there are only 4 here - don't we want to assign roles to all identities?

This script only does the role assignments that are scoped to the vnet RG. The others will be done by the RP via the cluster MSI.

>The operator names in the platform workload identity role set env var should be allowed to change, this hardcodes them

The script has to somehow know which role assignments go at the vnet RG scope. I don't see a better way than hard-coding them.

Also, this script is a temporary solution to accelerate local dev MIWI cluster creation. It'll be superseded by the automatic CLI stuff in a few months. So this doesn't need to be perfect or future-proof. Hard-coding the 4.14 operator names is fine IMO.

Reply:
Seeing Kipp's comment I agree that we are okay to use the hardcoded values in IF.
I believe it can be either way, assigning the desired role over the Network RG even when not needed shouldn't cause any problems other than anything that is identified during testing. So to be safe it is a good idea here to hardcode only the identities that will need interaction with Network RG.

Reply:
I see, thank you for the clarifications!